examples/react-native/screens/Dialer.js

```diff
@@ -0,0 +1,366 @@
+        this.setState({connected: true, status: 'connected'});
+      });
+
+      this.client.on('telnyx.error', error => {
+        console.log('error', error);
+      });
+
+      this.client.on('telnyx.notification', notification => {
+        switch (notification.type) {
+          case 'callUpdate':
+            return this._handleCallUpdate(notification.call);
+        }
+      });
+
+      this.client.on('telnyx.socket.open', () => {
+        console.log('Socket Open');
+      });
+      this.client.on('telnyx.socket.close', event => {
+        console.log('Socket Close', event);
+      });
+      this.client.on('telnyx.socket.error', error => {
+        console.log('error', error);
+
+        console.log('Socket Error');
+      });
+
+      this.client.connect();
+    }
+  }
+
+  makeCall() {
+    this.client.newCall({
+      destinationNumber: this.state.extension,
+      video: {facingMode: 'user'},
+    });
+  }
+
+  hangup() {
+    this.state.call.hangup();
+  }
+
+  toggleMic() {
+    this.setState({btnMicActive: !this.state.btnMicActive});
+    this.state.call.toggleAudioMute();
+  }
+
+  toggleCam() {
+    this.setState({btnCamActive: !this.state.btnCamActive});
+    this.state.call.toggleVideoMute();
+  }
+
+  toggleDeaf() {
+    this.setState({btnDeafActive: !this.state.btnDeafActive});
+    this.state.call.toggleDeaf();
+  }
+
+  switchCamera() {
+    this.state.call.switchCamera();
+  }
+
+  toggleSpeaker() {
+    this.setState({btnSpeakerActive: !this.state.btnSpeakerActive}, () => {
+      this.state.call.setSpeakerPhone(this.state.btnSpeakerActive);
+    });
+  }
+
+  handleDestination = extension => {
+    return this.setState({extension});
+  };
+
+  _handleCallUpdate(call) {
+    this.setState({status: call.state, cause: call.cause});
+
+    switch (call.state) {
+      case 'ringing': {
+        const {remoteCallerName, remoteCallerNumber} = call.options;
+        const caller = remoteCallerName || remoteCallerNumber;
+        Alert.alert(
+          'Inbound Call',
+          `Call from ${caller}`,
+          [
+            {
+              text: 'Reject',
+              onPress: () => call.hangup(),
+              style: 'cancel',
+            },
+            {
+              text: 'Answer',
+              onPress: () => call.answer(),
+            },
+          ],
+          {cancelable: false},
+        );
+        break;
+      }
+      case 'active':
+        this.setState({call, status: call.state, cause: call.cause});
+        break;
+      case 'destroy':
+        this.setState({call: null, status: call.state, cause: call.cause});
+        break;
+    }
+  }
+
+  _status() {
+    return this.state.connected ? 'Connected' : 'Disconnected';
+  }
+
+  _middle() {
+    if (this.state.call) {
+      let streamURL = null;
+      const {
+        options: {remoteStream = null, localStream = null},
+      } = this.state.call;
+      console.log('localStream', localStream, localStream.toURL());
```

Comment:
All those console.log should be present in the final version?

Reply:
It is only an example to user use as a base to build the own application